src/chat.rs

```diff
@@ -604,6 +606,34 @@ impl ChatId {
         Ok(())
     }
 
+    pub(crate) async fn add_encrypted_msg(
+        self,
+        context: &Context,
+        timestamp_sort: i64,
+    ) -> Result<()> {
+        let text = stock_str::messages_e2e_encrypted(context).await;
+        add_info_msg_with_cmd(
+            context,
+            self,
+            &text,
+            SystemMessage::ChatE2ee,
+            timestamp_sort,
+            None,
+            None,
+            None,
+            None,
+        )
+        .await?;
+        Ok(())
+    }
+
+    /// Returns true if the chat is encrypted, plain chat.
+    pub(crate) async fn is_encrypted_and_not_special(self, context: &Context) -> Result<bool> {
+        let chat = Chat::load_from_db(context, self).await?;
+        let res = chat.is_encrypted(context).await? && !chat.is_device_talk();
```

Comment:
Performance heads-up(non-blocking): `is_encrypted_and_not_special()` does up to 3 database calls:
- `Chat::load_from_db`,
- and then inside `is_encrypted()`:
  - `get_chat_contacts()`
  - `Contact::get_by_id()`

It's only called when creating a new chat, which is not that often, so, it's fine. It's a slight footgun because someone might use it in a performance-sensitive context, but it's probably fine.

Reply:
yip, but i do not see an approach with less calls. but worth a comment

Reply:
There are things we could do, like custom SQL statements, but they are not worth it here, and would need benchmarking.